sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/SqlDelightEnvironment.kt

```diff
@@ -170,6 +181,29 @@ class SqlDelightEnvironment(
       logger("----- END ${it.name} in $timeTaken ms ------")
     }
 
+    if (properties.dependencies.isEmpty()) {
```

Comment:
This is needed to only create the systems tables once, otherwise there will be a R8 error when multiple dbs create the same class.

Reply:
its possible in a single project to have multiple databases which aren't connected, which if I understand correctly will no longer be possible because they will generate conflicting classes

Reply:
unless they use the package name of the db which would disambiguate

Reply:
yea they all share the same `sqlite.schema` package name so it will break

Reply:
Ahhhh

Reply:
Well, we simply could use the user defined package name, I just find it somehow "wrong".

Reply:
an alternative would be to publish a runtime artifact with these classes, and only depend on them if the user needs them. Which is....a lot

Reply:
Another alternative would be to not generate common shared classes at all, and always generate a query result type.

So:

```
selectFromSystemTable:
SELECT *
FROM sqlite_sequence;
```

would generate:

```
data class SelectFromSystemTable(
...
)
```

I think this is actually my preference, it should require very little change because somewhere in the compiler theres a "needs custom type" on NamedQuery or something that we can just return to true and it will do exactly this

Reply:
Okay, this is the easiest way :D I didn't do this is the past because the resulting system tables will have the user defined package. But if you are fine with this behavior, I could do this.

Reply:
Implemented, it will only generated classes if the user queries them.